Subject: [PATCH] Publish OV androidx.annotation 1.8.0
---
Index: gradle/wrapper/gradle-wrapper.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/gradle/wrapper/gradle-wrapper.properties b/gradle/wrapper/gradle-wrapper.properties
--- a/gradle/wrapper/gradle-wrapper.properties	(revision 06253a1d13b8f3530c34bcd8a4dab6904dffe2c1)
+++ b/gradle/wrapper/gradle-wrapper.properties	(revision aa7f42ac73fdcc018e7947331b7e0f437c3a8b47)
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=../../../../tools/external/gradle/gradle-8.7-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-bin.zip
 distributionSha256Sum=544c35d6bd849ae8a5ed0bcea39ba677dc40f49df7d1835561582da2009b961d
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
Index: settings.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/settings.gradle b/settings.gradle
--- a/settings.gradle	(revision 06253a1d13b8f3530c34bcd8a4dab6904dffe2c1)
+++ b/settings.gradle	(revision 3b23fa585e65158101d741ea69b9428284608296)
@@ -11,6 +11,9 @@
             maven {
                 url = "https://plugins.gradle.org/m2/"
             }
+            maven {
+                url = "https://mirrors.tencent.com/nexus/repository/maven-public"
+            }
         }
     }
 }
@@ -379,10 +382,10 @@
 includeProject(":autofill:autofill", [BuildType.MAIN])
 includeProject(":benchmark:benchmark-benchmark", "benchmark/benchmark", [BuildType.MAIN, BuildType.COMPOSE])
 includeProject(":benchmark:benchmark-common")
-includeProject(":benchmark:benchmark-darwin", [BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":benchmark:benchmark-darwin-core", [BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":benchmark:benchmark-darwin-samples", [BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":benchmark:benchmark-darwin-gradle-plugin", [BuildType.INFRAROGUE, BuildType.KMP])
+includeProject(":benchmark:benchmark-darwin", [BuildType.INFRAROGUE])
+includeProject(":benchmark:benchmark-darwin-core", [BuildType.INFRAROGUE])
+includeProject(":benchmark:benchmark-darwin-samples", [BuildType.INFRAROGUE])
+includeProject(":benchmark:benchmark-darwin-gradle-plugin", [BuildType.INFRAROGUE])
 includeProject(":benchmark:benchmark-gradle-plugin", "benchmark/gradle-plugin", [BuildType.MAIN])
 includeProject(":benchmark:benchmark-baseline-profile-gradle-plugin", "benchmark/baseline-profile-gradle-plugin", [BuildType.MAIN, BuildType.COMPOSE])
 includeProject(":benchmark:benchmark-junit4")
@@ -453,11 +456,11 @@
 includeProject(":car:app:app-samples:showcase-mobile", "car/app/app-samples/showcase/mobile", [BuildType.MAIN])
 includeProject(":car:app:app-testing", [BuildType.MAIN])
 includeProject(":cardview:cardview", [BuildType.MAIN])
-includeProject(":collection:collection", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":collection:collection-benchmark", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":collection:collection-benchmark-kmp", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":collection:collection-ktx", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":collection:integration-tests:testapp", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
+includeProject(":collection:collection", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":collection:collection-benchmark", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":collection:collection-benchmark-kmp", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":collection:collection-ktx", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":collection:integration-tests:testapp", [BuildType.MAIN, BuildType.INFRAROGUE])
 includeProject(":compose:animation", [BuildType.COMPOSE])
 includeProject(":compose:animation:animation", [BuildType.COMPOSE])
 includeProject(":compose:animation:animation-lint", [BuildType.COMPOSE])
@@ -479,9 +482,9 @@
 includeProject(":compose:compiler:compiler-hosted:runtime-tests", [BuildType.COMPOSE])
 includeProject(":compose:compiler:compiler-hosted:integration-tests:kotlin-compiler-repackaged", [BuildType.COMPOSE])
 includeProject(":compose:compiler:compiler-hosted:integration-tests:protobuf-test-classes", [BuildType.COMPOSE])
-includeProject(":compose:desktop", [BuildType.COMPOSE, BuildType.KMP])
-includeProject(":compose:desktop:desktop", [BuildType.COMPOSE, BuildType.KMP])
-includeProject(":compose:desktop:desktop:desktop-samples", "compose/desktop/desktop/samples", [BuildType.COMPOSE, BuildType.KMP])
+includeProject(":compose:desktop", [BuildType.COMPOSE])
+includeProject(":compose:desktop:desktop", [BuildType.COMPOSE])
+includeProject(":compose:desktop:desktop:desktop-samples", "compose/desktop/desktop/samples", [BuildType.COMPOSE])
 includeProject(":compose:foundation", [BuildType.COMPOSE])
 includeProject(":compose:foundation:foundation", [BuildType.COMPOSE])
 includeProject(":compose:foundation:foundation-benchmark", "compose/foundation/foundation/benchmark", [BuildType.COMPOSE])
@@ -540,7 +543,7 @@
 includeProject(":compose:material3:material3:integration-tests:material3-catalog", [BuildType.COMPOSE])
 includeProject(":compose:material:material:material-samples", "compose/material/material/samples", [BuildType.COMPOSE])
 includeProject(":compose:material3:material3:material3-samples", "compose/material3/material3/samples", [BuildType.COMPOSE])
-includeProject(":compose:runtime:runtime", [BuildType.COMPOSE, BuildType.KMP])
+includeProject(":compose:runtime:runtime", [BuildType.COMPOSE])
 includeProject(":compose:runtime:runtime-lint", [BuildType.COMPOSE])
 includeProject(":compose:runtime:runtime-livedata", [BuildType.COMPOSE])
 includeProject(":compose:runtime:runtime-livedata:runtime-livedata-samples", "compose/runtime/runtime-livedata/samples", [BuildType.COMPOSE])
@@ -643,20 +646,20 @@
 includeProject(":cursoradapter:cursoradapter", [BuildType.MAIN])
 includeProject(":customview:customview", [BuildType.MAIN])
 includeProject(":customview:customview-poolingcontainer", [BuildType.MAIN, BuildType.COMPOSE])
-includeProject(":datastore:datastore", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-benchmark", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-core", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-core-okio", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
+includeProject(":datastore:datastore", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-benchmark", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-core", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-core-okio", [BuildType.MAIN, BuildType.INFRAROGUE])
 includeProject(":datastore:datastore-compose-samples", [BuildType.COMPOSE])
-includeProject(":datastore:datastore-preferences", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-preferences-core", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-preferences-proto", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-preferences-rxjava2", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-preferences-rxjava3", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-proto", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-rxjava2", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-rxjava3", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
-includeProject(":datastore:datastore-sampleapp", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
+includeProject(":datastore:datastore-preferences", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-preferences-core", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-preferences-proto", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-preferences-rxjava2", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-preferences-rxjava3", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-proto", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-rxjava2", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-rxjava3", [BuildType.MAIN, BuildType.INFRAROGUE])
+includeProject(":datastore:datastore-sampleapp", [BuildType.MAIN, BuildType.INFRAROGUE])
 includeProject(":documentfile:documentfile", [BuildType.MAIN])
 includeProject(":draganddrop:draganddrop", [BuildType.MAIN])
 includeProject(":draganddrop:integration-tests:sampleapp", [BuildType.MAIN])
@@ -813,7 +816,7 @@
 includeProject(":navigation:navigation-ui", [BuildType.MAIN, BuildType.FLAN])
 includeProject(":navigation:navigation-ui-ktx", [BuildType.MAIN, BuildType.FLAN])
 includeProject(":paging:integration-tests:testapp", [BuildType.MAIN])
-includeProject(":paging:paging-common", [BuildType.MAIN, BuildType.COMPOSE, BuildType.INFRAROGUE, BuildType.KMP])
+includeProject(":paging:paging-common", [BuildType.MAIN, BuildType.COMPOSE, BuildType.INFRAROGUE])
 includeProject(":paging:paging-common-ktx", [BuildType.MAIN, BuildType.COMPOSE])
 includeProject(":paging:paging-compose", [BuildType.COMPOSE])
 includeProject(":paging:paging-compose:paging-compose-samples", "paging/paging-compose/samples", [BuildType.COMPOSE])
@@ -825,7 +828,7 @@
 includeProject(":paging:paging-rxjava2-ktx", [BuildType.MAIN])
 includeProject(":paging:paging-rxjava3", [BuildType.MAIN])
 includeProject(":paging:paging-samples", "paging/samples", [BuildType.MAIN, BuildType.COMPOSE])
-includeProject(":paging:paging-testing", [BuildType.MAIN, BuildType.COMPOSE, BuildType.INFRAROGUE, BuildType.KMP])
+includeProject(":paging:paging-testing", [BuildType.MAIN, BuildType.COMPOSE, BuildType.INFRAROGUE])
 includeProject(":palette:palette", [BuildType.MAIN])
 includeProject(":palette:palette-ktx", [BuildType.MAIN])
 includeProject(":pdf:pdf-viewer", [BuildType.MAIN])
@@ -880,28 +883,28 @@
 includeProject(":room:integration-tests:room-incremental-annotation-processing", "room/integration-tests/incremental-annotation-processing", [BuildType.MAIN])
 includeProject(":room:integration-tests:room-testapp", "room/integration-tests/testapp", [BuildType.MAIN])
 includeProject(":room:integration-tests:room-testapp-autovalue", "room/integration-tests/autovaluetestapp", [BuildType.MAIN])
-includeProject(":room:integration-tests:room-testapp-kotlin", "room/integration-tests/kotlintestapp", [BuildType.MAIN, BuildType.KMP])
-includeProject(":room:integration-tests:room-testapp-multiplatform", "room/integration-tests/multiplatformtestapp", [BuildType.MAIN, BuildType.KMP, BuildType.INFRAROGUE])
+includeProject(":room:integration-tests:room-testapp-kotlin", "room/integration-tests/kotlintestapp", [BuildType.MAIN])
+includeProject(":room:integration-tests:room-testapp-multiplatform", "room/integration-tests/multiplatformtestapp", [BuildType.MAIN, BuildType.INFRAROGUE])
 includeProject(":room:integration-tests:room-testapp-noappcompat", "room/integration-tests/noappcompattestapp", [BuildType.MAIN])
 includeProject(":room:room-benchmark", "room/benchmark", [BuildType.MAIN])
-includeProject(":room:room-common", [BuildType.MAIN, BuildType.COMPOSE, BuildType.KMP, BuildType.INFRAROGUE])
-includeProject(":room:room-compiler", [BuildType.MAIN, BuildType.COMPOSE, BuildType.KMP])
-includeProject(":room:room-compiler-processing", [BuildType.MAIN, BuildType.COMPOSE, BuildType.FLAN, BuildType.KMP])
+includeProject(":room:room-common", [BuildType.MAIN, BuildType.COMPOSE, BuildType.INFRAROGUE])
+includeProject(":room:room-compiler", [BuildType.MAIN, BuildType.COMPOSE])
+includeProject(":room:room-compiler-processing", [BuildType.MAIN, BuildType.COMPOSE, BuildType.FLAN])
 includeProject(":room:room-compiler-processing-testing", [BuildType.MAIN, BuildType.COMPOSE, BuildType.FLAN])
 includeProject(":room:room-external-antlr", [BuildType.MAIN])
 includeProject(":room:room-guava", [BuildType.MAIN])
 includeProject(":room:room-gradle-plugin", [BuildType.MAIN])
 includeProject(":room:room-ktx", [BuildType.MAIN, BuildType.COMPOSE])
-includeProject(":room:room-migration", [BuildType.MAIN, BuildType.COMPOSE, BuildType.KMP, BuildType.INFRAROGUE])
+includeProject(":room:room-migration", [BuildType.MAIN, BuildType.COMPOSE, BuildType.INFRAROGUE])
 includeProject(":room:room-paging", [BuildType.MAIN, BuildType.COMPOSE])
 includeProject(":room:room-paging-guava", [BuildType.MAIN])
 includeProject(":room:room-paging-rxjava2", [BuildType.MAIN])
 includeProject(":room:room-paging-rxjava3", [BuildType.MAIN])
-includeProject(":room:room-runtime", [BuildType.MAIN, BuildType.COMPOSE, BuildType.KMP, BuildType.INFRAROGUE])
+includeProject(":room:room-runtime", [BuildType.MAIN, BuildType.COMPOSE, BuildType.INFRAROGUE])
 includeProject(":room:room-runtime-lint", [BuildType.MAIN, BuildType.COMPOSE])
 includeProject(":room:room-rxjava2", [BuildType.MAIN])
 includeProject(":room:room-rxjava3", [BuildType.MAIN])
-includeProject(":room:room-testing", [BuildType.MAIN, BuildType.KMP, BuildType.INFRAROGUE])
+includeProject(":room:room-testing", [BuildType.MAIN, BuildType.INFRAROGUE])
 includeProject(":safeparcel:safeparcel", [BuildType.MAIN])
 includeProject(":safeparcel:safeparcel-processor", [BuildType.MAIN])
 includeProject(":savedstate:savedstate", [BuildType.MAIN, BuildType.COMPOSE, BuildType.FLAN])
@@ -925,12 +928,12 @@
 includeProject(":slice:slice-view", [BuildType.MAIN])
 includeProject(":slidingpanelayout:slidingpanelayout", [BuildType.MAIN, BuildType.FLAN])
 includeProject(":slidingpanelayout:slidingpanelayout-testapp", [BuildType.MAIN])
-includeProject(":sqlite:integration-tests:driver-conformance-test", [BuildType.MAIN, BuildType.KMP, BuildType.INFRAROGUE])
+includeProject(":sqlite:integration-tests:driver-conformance-test", [BuildType.MAIN, BuildType.INFRAROGUE])
 includeProject(":sqlite:integration-tests:inspection-room-testapp", [BuildType.MAIN])
 includeProject(":sqlite:integration-tests:inspection-sqldelight-testapp", [BuildType.MAIN])
-includeProject(":sqlite:sqlite", [BuildType.MAIN, BuildType.COMPOSE, BuildType.KMP, BuildType.INFRAROGUE])
-includeProject(":sqlite:sqlite-framework", [BuildType.MAIN, BuildType.COMPOSE, BuildType.KMP, BuildType.INFRAROGUE])
-includeProject(":sqlite:sqlite-bundled", [BuildType.MAIN, BuildType.KMP, BuildType.INFRAROGUE])
+includeProject(":sqlite:sqlite", [BuildType.MAIN, BuildType.COMPOSE, BuildType.INFRAROGUE])
+includeProject(":sqlite:sqlite-framework", [BuildType.MAIN, BuildType.COMPOSE, BuildType.INFRAROGUE])
+includeProject(":sqlite:sqlite-bundled", [BuildType.MAIN, BuildType.INFRAROGUE])
 includeProject(":sqlite:sqlite-inspection", [BuildType.MAIN])
 includeProject(":sqlite:sqlite-ktx", [BuildType.MAIN])
 includeProject(":startup:integration-tests:first-library", [BuildType.MAIN])
@@ -1056,7 +1059,7 @@
 includeProject(":window:integration-tests:macrobenchmark", [BuildType.MAIN, BuildType.WINDOW])
 includeProject(":window:sidecar:sidecar", [BuildType.MAIN, BuildType.COMPOSE, BuildType.FLAN, BuildType.WINDOW])
 includeProject(":window:window-java", [BuildType.MAIN, BuildType.COMPOSE, BuildType.FLAN, BuildType.WINDOW])
-includeProject(":window:window-core", [BuildType.MAIN, BuildType.COMPOSE, BuildType.FLAN, BuildType.INFRAROGUE, BuildType.KMP, BuildType.WINDOW])
+includeProject(":window:window-core", [BuildType.MAIN, BuildType.COMPOSE, BuildType.FLAN, BuildType.INFRAROGUE, BuildType.WINDOW])
 includeProject(":window:window-rxjava2", [BuildType.MAIN, BuildType.WINDOW])
 includeProject(":window:window-rxjava3", [BuildType.MAIN, BuildType.WINDOW])
 includeProject(":window:window-demos:demo", [BuildType.MAIN, BuildType.COMPOSE, BuildType.FLAN, BuildType.WINDOW])
@@ -1124,7 +1127,7 @@
 /////////////////////////////
 
 includeProject(":internal-testutils-common", "testutils/testutils-common", [BuildType.MAIN, BuildType.COMPOSE, BuildType.FLAN])
-includeProject(":internal-testutils-datastore", "testutils/testutils-datastore", [BuildType.MAIN, BuildType.INFRAROGUE, BuildType.KMP])
+includeProject(":internal-testutils-datastore", "testutils/testutils-datastore", [BuildType.MAIN, BuildType.INFRAROGUE])
 includeProject(":internal-testutils-runtime", "testutils/testutils-runtime", [BuildType.MAIN, BuildType.FLAN, BuildType.COMPOSE, BuildType.MEDIA, BuildType.WEAR])
 includeProject(":internal-testutils-appcompat", "testutils/testutils-appcompat", [BuildType.MAIN])
 includeProject(":internal-testutils-espresso", "testutils/testutils-espresso", [BuildType.MAIN, BuildType.COMPOSE])
Index: buildSrc/repos.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/repos.gradle b/buildSrc/repos.gradle
--- a/buildSrc/repos.gradle	(revision aa7f42ac73fdcc018e7947331b7e0f437c3a8b47)
+++ b/buildSrc/repos.gradle	(revision 3b23fa585e65158101d741ea69b9428284608296)
@@ -63,6 +63,9 @@
             }
         }
     }
+    handler.maven {
+        url = "https://mirrors.tencent.com/nexus/repository/maven-public"
+    }
     if (System.getenv("ALLOW_PUBLIC_REPOS") != null || System.getProperty("ALLOW_PUBLIC_REPOS") != null) {
         handler.mavenCentral()
         handler.google()
Index: gradle/libs.versions.toml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
--- a/gradle/libs.versions.toml	(revision aa7f42ac73fdcc018e7947331b7e0f437c3a8b47)
+++ b/gradle/libs.versions.toml	(revision 2eb64f15e98c0135fa177dbbd9efef18ee59c830)
@@ -41,10 +41,10 @@
 jcodec = "0.2.5"
 kotlin17 = "1.7.10"
 kotlin18 = "1.8.22"
-kotlin19 = "1.9.23"
-kotlin = "1.9.23"
+kotlin19 = "2.0.21-KBA-001"
+kotlin = "2.0.21-KBA-001"
 kotlinBenchmark = "0.4.8"
-kotlinNative = "1.9.23"
+kotlinNative = "2.0.21-KBA-001"
 kotlinCompileTesting = "1.4.9"
 kotlinCoroutines = "1.7.3"
 kotlinSerialization = "1.6.3"
@@ -53,7 +53,7 @@
 ktlint = "0.49.1"
 leakcanary = "2.13"
 media3 = "1.1.0"
-metalava = "1.0.0-alpha10"
+metalava = "1.0.0-alpha11"
 mockito = "2.25.0"
 moshi = "1.13.0"
 protobuf = "3.22.3"
Index: playground-common/playground-plugin/settings.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/playground-common/playground-plugin/settings.gradle b/playground-common/playground-plugin/settings.gradle
--- a/playground-common/playground-plugin/settings.gradle	(revision aa7f42ac73fdcc018e7947331b7e0f437c3a8b47)
+++ b/playground-common/playground-plugin/settings.gradle	(revision 3b23fa585e65158101d741ea69b9428284608296)
@@ -21,6 +21,9 @@
         gradlePluginPortal().content {
             it.includeModule("org.jetbrains.kotlin.jvm", "org.jetbrains.kotlin.jvm.gradle.plugin")
         }
+        maven {
+            url = "https://mirrors.tencent.com/nexus/repository/maven-public"
+        }
     }
 }
 
@@ -36,6 +39,9 @@
                     "com.github.johnrengelman.shadow.gradle.plugin")
             it.includeModule("com.github.johnrengelman", "shadow")
         }
+        maven {
+            url = "https://mirrors.tencent.com/nexus/repository/maven-public"
+        }
     }
 }
 
Index: buildSrc/private/src/main/kotlin/androidx/build/AndroidXMultiplatformExtension.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXMultiplatformExtension.kt b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXMultiplatformExtension.kt
--- a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXMultiplatformExtension.kt	(revision a47a0f45e33607c5d311cd28db8852f0cd8ac249)
+++ b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXMultiplatformExtension.kt	(revision e07aaca703658a75278aa8ddec73ebbb2b2384ae)
@@ -492,6 +492,20 @@
         }
     }
 
+    @JvmOverloads
+    fun ohosArm64(block: Action<KotlinNativeTarget>? = null): KotlinNativeTarget? {
+        supportedPlatforms.add(PlatformIdentifier.OHOS_ARM_64)
+        return if (project.enableOhosNative()) {
+            if (block == null) kotlinExtension.ohosArm64() else kotlinExtension.ohosArm64(block)
+        } else {
+            null
+        }
+    }
+
+    @JvmOverloads
+    fun ohosNative(block: Action<KotlinNativeTarget>? = null): List<KotlinNativeTarget> =
+        listOfNotNull(ohosArm64(block))
+
     @JvmOverloads
     fun linux(block: Action<KotlinNativeTarget>? = null): List<KotlinNativeTarget> {
         return listOfNotNull(
Index: buildSrc/public/src/main/kotlin/androidx/build/KmpPlatforms.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/public/src/main/kotlin/androidx/build/KmpPlatforms.kt b/buildSrc/public/src/main/kotlin/androidx/build/KmpPlatforms.kt
--- a/buildSrc/public/src/main/kotlin/androidx/build/KmpPlatforms.kt	(revision a47a0f45e33607c5d311cd28db8852f0cd8ac249)
+++ b/buildSrc/public/src/main/kotlin/androidx/build/KmpPlatforms.kt	(revision 8af4185851655609d8f28542da0d6c1c9b45e294)
@@ -37,7 +37,8 @@
     MAC,
     LINUX,
     DESKTOP,
-    ANDROID_NATIVE;
+    ANDROID_NATIVE,
+    OHOS_NATIVE;
 
     companion object {
         /** Target platform groups which require native compilation (e.g. LLVM). */
@@ -49,7 +50,7 @@
          * Do *not* enable [JS] unless you have read and understand this:
          * https://blog.jetbrains.com/kotlin/2021/10/important-ua-parser-js-exploit-and-kotlin-js/
          */
-        val enabledByDefault = listOf(JVM, DESKTOP, MAC, LINUX, ANDROID_NATIVE)
+        val enabledByDefault = listOf(JVM, DESKTOP, MAC, LINUX, ANDROID_NATIVE, OHOS_NATIVE)
     }
 }
 
@@ -71,6 +72,7 @@
     IOS_SIMULATOR_ARM_64("iossimulatorarm64", PlatformGroup.MAC),
     IOS_X_64("iosx64", PlatformGroup.MAC),
     IOS_ARM_64("iosarm64", PlatformGroup.MAC),
+    OHOS_ARM_64("ohosarm64", PlatformGroup.OHOS_NATIVE),
     DESKTOP("desktop", PlatformGroup.JVM);
 
     companion object {
@@ -124,6 +126,9 @@
 fun Project.enableAndroidNative(): Boolean =
     enabledKmpPlatforms.contains(PlatformGroup.ANDROID_NATIVE)
 
+fun Project.enableOhosNative(): Boolean =
+    enabledKmpPlatforms.contains(PlatformGroup.OHOS_NATIVE)
+
 fun Project.enableMac(): Boolean =
     enabledKmpPlatforms.contains(PlatformGroup.MAC) && HostManager.hostIsMac
 
Index: buildSrc/private/src/main/kotlin/androidx/build/MavenUploadHelper.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/private/src/main/kotlin/androidx/build/MavenUploadHelper.kt b/buildSrc/private/src/main/kotlin/androidx/build/MavenUploadHelper.kt
--- a/buildSrc/private/src/main/kotlin/androidx/build/MavenUploadHelper.kt	(revision 550101f471fcba1366f848be94e131a3309552fc)
+++ b/buildSrc/private/src/main/kotlin/androidx/build/MavenUploadHelper.kt	(revision e5ad34b1a65f00a65e65e1f268c0454fe810db26)
@@ -30,10 +30,12 @@
 import java.io.StringWriter
 import org.dom4j.Element
 import org.dom4j.io.XMLWriter
+import org.gradle.api.Action
 import org.gradle.api.GradleException
 import org.gradle.api.Project
 import org.gradle.api.XmlProvider
 import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.repositories.PasswordCredentials
 import org.gradle.api.component.ComponentWithVariants
 import org.gradle.api.component.SoftwareComponent
 import org.gradle.api.component.SoftwareComponentFactory
@@ -101,7 +103,7 @@
 private fun Project.validateTaskIsRegistered(taskName: String) =
     tasks.findByName(taskName) ?: throw GradleException(
         "Project $name is configured for publishing, but a '$taskName' task was never " +
-        "registered. This is likely a bug in AndroidX plugin configuration."
+            "registered. This is likely a bug in AndroidX plugin configuration."
     )
 
 private fun Project.releaseTaskShouldBeRegistered(extension: AndroidXExtension): Boolean {
@@ -159,6 +161,17 @@
 
     configure<PublishingExtension> {
         repositories { it.maven { repo -> repo.setUrl(getRepositoryDirectory()) } }
+        repositories {
+            it.maven { repo ->
+                repo.name = "RemoteMaven"
+                repo.setUrl(findProperty("maven.remote.url").toString())
+                val passwordCredentials = Action<PasswordCredentials> { credentials ->
+                    credentials.username = findProperty("maven.remote.username")?.toString()
+                    credentials.password = findProperty("maven.remote.password")?.toString()
+                }
+                repo.credentials(passwordCredentials)
+            }
+        }
         publications {
             if (appliesJavaGradlePluginPlugin()) {
                 // The 'java-gradle-plugin' will also add to the 'pluginMaven' publication
@@ -199,10 +212,10 @@
                     "." + project.name.toString().replace('-', '.') + ".anchor"
                 val unpackedStubAarTask =
                     tasks.register("unpackedStubAar", UnpackedStubAarTask::class.java) { aarTask ->
-                    aarTask.aarPackage.set(stubNamespace)
-                    aarTask.minSdkVersion.set(baseExtension.defaultConfig.minSdk)
-                    aarTask.outputDir.set(buildDir.dir("intermediates/stub-aar"))
-                }
+                        aarTask.aarPackage.set(stubNamespace)
+                        aarTask.minSdkVersion.set(baseExtension.defaultConfig.minSdk)
+                        aarTask.outputDir.set(buildDir.dir("intermediates/stub-aar"))
+                    }
                 val stubAarTask = tasks.register("stubAar", Zip::class.java) { zipTask ->
                     zipTask.from(unpackedStubAarTask.flatMap { it.outputDir })
                     zipTask.destinationDirectory.set(buildDir.dir("outputs"))
@@ -271,24 +284,24 @@
     val document = parseXml(pom, namespaceUris)
 
     // For each <dependencies> element, sort the contained elements in-place.
-    document.rootElement.selectNodes("ns:dependencies").filterIsInstance<Element>().forEach {
-        element ->
-        val deps = element.elements()
-        val sortedDeps = deps.toSortedSet(compareBy { it.stringValue }).toList()
+    document.rootElement.selectNodes("ns:dependencies").filterIsInstance<Element>()
+        .forEach { element ->
+            val deps = element.elements()
+            val sortedDeps = deps.toSortedSet(compareBy { it.stringValue }).toList()
 
-        // Content contains formatting nodes, so to avoid modifying those we replace
-        // each element with the sorted element from its respective index. Note this
-        // will not move adjacent elements, so any comments would remain in their
-        // original order.
-        element.content().replaceAll {
-            val index = deps.indexOf(it)
-            if (index >= 0) {
-                sortedDeps[index]
-            } else {
-                it
-            }
-        }
-    }
+            // Content contains formatting nodes, so to avoid modifying those we replace
+            // each element with the sorted element from its respective index. Note this
+            // will not move adjacent elements, so any comments would remain in their
+            // original order.
+            element.content().replaceAll {
+                val index = deps.indexOf(it)
+                if (index >= 0) {
+                    sortedDeps[index]
+                } else {
+                    it
+                }
+            }
+        }
 
     // Write to string. Note that this does not preserve the original indent level, but it
     // does preserve line breaks -- not that any of this matters for client XML parsing.
@@ -395,7 +408,7 @@
                                 // Include sources artifact we built and root artifacts from kotlin
                                 // plugin.
                                 return (sourcesComponents.flatMap { it.usages } +
-                                        kotlinComponent.usages)
+                                    kotlinComponent.usages)
                                     .toMutableSet()
                             }
 
@@ -482,8 +495,8 @@
     val strippedGroupId = mavenGroup.group.substringAfterLast(".")
     if (
         !extension.bypassCoordinateValidation &&
-            mavenGroup.group.startsWith("androidx") &&
-            !name.startsWith(strippedGroupId)
+        mavenGroup.group.startsWith("androidx") &&
+        !name.startsWith(strippedGroupId)
     ) {
         throw Exception("Your artifactId must start with '$strippedGroupId'. (currently is $name)")
     }
Index: buildSrc/private/src/main/kotlin/androidx/build/AndroidXExtension.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXExtension.kt b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXExtension.kt
--- a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXExtension.kt	(revision 5dbfb37a53a64cf6b582531fd4282d8fb90edc3a)
+++ b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXExtension.kt	(revision 2eb64f15e98c0135fa177dbbd9efef18ee59c830)
@@ -234,13 +234,13 @@
                         ")"
                 )
             } else {
-                verifyVersionExtraFormat(mavenVersion)
+//                verifyVersionExtraFormat(mavenVersion)
                 version = mavenVersion
             }
         } else {
             projectDirectlySpecifiesMavenVersion = false
             if (groupVersion != null) {
-                verifyVersionExtraFormat(groupVersion)
+//                verifyVersionExtraFormat(groupVersion)
                 version = groupVersion
             } else {
                 return
@@ -255,28 +255,58 @@
 
     private fun verifyVersionExtraFormat(version: Version) {
         val ALLOWED_EXTRA_PREFIXES = listOf("-alpha", "-beta", "-rc", "-dev", "-SNAPSHOT")
-        val extra = version.extra
-        if (extra != null) {
-            if (!version.isSnapshot() && project.isVersionExtraCheckEnabled()) {
-                if (ALLOWED_EXTRA_PREFIXES.any { extra.startsWith(it) }) {
-                    for (potentialPrefix in ALLOWED_EXTRA_PREFIXES) {
-                        if (extra.startsWith(potentialPrefix)) {
-                            val secondExtraPart = extra.removePrefix(potentialPrefix)
-                            if (secondExtraPart.toIntOrNull() == null) {
-                                throw IllegalArgumentException(
-                                    "Version $version is not" +
-                                        " a properly formatted version, please ensure that " +
-                                        "$potentialPrefix is followed by a number only"
-                                )
-                            }
-                        }
-                    }
-                } else {
-                    throw IllegalArgumentException(
-                        "Version $version is not a proper " +
-                            "version, version suffixes following major.minor.patch should " +
-                            "be one of ${ALLOWED_EXTRA_PREFIXES.joinToString(", ")}"
-                    )
+        if (project.isSubversionCheckEnabled()) {
+            val subVersion = version.subVersion()
+                ?: throw IllegalArgumentException("Version $version is not a properly formatted version, version should be something like major.minor.patch-x.y.z")
+            val extra = subVersion.extra
+            if (extra != null) {
+                if (!version.isSnapshot() && project.isVersionExtraCheckEnabled()) {
+                    if (ALLOWED_EXTRA_PREFIXES.any { extra.startsWith(it) }) {
+                        for (potentialPrefix in ALLOWED_EXTRA_PREFIXES) {
+                            if (extra.startsWith(potentialPrefix)) {
+                                val secondExtraPart = extra.removePrefix(potentialPrefix)
+                                if (secondExtraPart.toIntOrNull() == null) {
+                                    throw IllegalArgumentException(
+                                        "Version $version is not" +
+                                            " a properly formatted version, please ensure that " +
+                                            "$potentialPrefix is followed by a number only"
+                                    )
+                                }
+                            }
+                        }
+                    } else {
+                        throw IllegalArgumentException(
+                            "Version $version is not a proper " +
+                                "version, version suffixes following major.minor.patch-x.y.z should " +
+                                "be one of ${ALLOWED_EXTRA_PREFIXES.joinToString(", ")}"
+                        )
+                    }
+                }
+            }
+        } else {
+            val extra = version.extra
+            if (extra != null) {
+                if (!version.isSnapshot() && project.isVersionExtraCheckEnabled()) {
+                    if (ALLOWED_EXTRA_PREFIXES.any { extra.startsWith(it) }) {
+                        for (potentialPrefix in ALLOWED_EXTRA_PREFIXES) {
+                            if (extra.startsWith(potentialPrefix)) {
+                                val secondExtraPart = extra.removePrefix(potentialPrefix)
+                                if (secondExtraPart.toIntOrNull() == null) {
+                                    throw IllegalArgumentException(
+                                        "Version $version is not" +
+                                            " a properly formatted version, please ensure that " +
+                                            "$potentialPrefix is followed by a number only"
+                                    )
+                                }
+                            }
+                        }
+                    } else {
+                        throw IllegalArgumentException(
+                            "Version $version is not a proper " +
+                                "version, version suffixes following major.minor.patch should " +
+                                "be one of ${ALLOWED_EXTRA_PREFIXES.joinToString(", ")}"
+                        )
+                    }
                 }
             }
         }
Index: buildSrc/private/src/main/kotlin/androidx/build/AndroidXGradleProperties.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXGradleProperties.kt b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXGradleProperties.kt
--- a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXGradleProperties.kt	(revision 5dbfb37a53a64cf6b582531fd4282d8fb90edc3a)
+++ b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXGradleProperties.kt	(revision 2f9bf2b8adda49cb44020bd7ea671eb78cd3ca7b)
@@ -50,6 +50,8 @@
  */
 const val VERSION_EXTRA_CHECK_ENABLED = "androidx.versionExtraCheckEnabled"
 
+const val SUBVERSION_CHECK_ENABLED = "androidx.subVersionCheckEnabled"
+
 /** Validate the project structure against Jetpack guidelines */
 const val VALIDATE_PROJECT_STRUCTURE = "androidx.validateProjectStructure"
 
@@ -229,6 +231,9 @@
 fun Project.isVersionExtraCheckEnabled(): Boolean =
     findBooleanProperty(VERSION_EXTRA_CHECK_ENABLED) ?: true
 
+fun Project.isSubversionCheckEnabled(): Boolean =
+    findBooleanProperty(SUBVERSION_CHECK_ENABLED) ?: false
+
 /** Validate the project structure against Jetpack guidelines */
 fun Project.isValidateProjectStructureEnabled(): Boolean =
     findBooleanProperty(VALIDATE_PROJECT_STRUCTURE) ?: true
Index: buildSrc/public/src/main/kotlin/androidx/build/Version.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/public/src/main/kotlin/androidx/build/Version.kt b/buildSrc/public/src/main/kotlin/androidx/build/Version.kt
--- a/buildSrc/public/src/main/kotlin/androidx/build/Version.kt	(revision 5dbfb37a53a64cf6b582531fd4282d8fb90edc3a)
+++ b/buildSrc/public/src/main/kotlin/androidx/build/Version.kt	(revision 2f9bf2b8adda49cb44020bd7ea671eb78cd3ca7b)
@@ -36,6 +36,8 @@
         else null
     )
 
+    fun subVersion(): Version? = extra?.takeIf { it.startsWith("-") }?.let { Version(it.drop(1)) }
+
     fun isPatch(): Boolean = patch != 0
 
     fun isSnapshot(): Boolean = "-SNAPSHOT" == extra
Index: annotation/annotation/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/annotation/annotation/build.gradle b/annotation/annotation/build.gradle
--- a/annotation/annotation/build.gradle	(revision 2f9bf2b8adda49cb44020bd7ea671eb78cd3ca7b)
+++ b/annotation/annotation/build.gradle	(revision 566a9f8d01b29a076829c9cf2d59e1fb54408980)
@@ -21,6 +21,7 @@
     mac()
     linux()
     ios()
+    ohosNative()
 
     defaultPlatform(PlatformIdentifier.JVM)
 
Index: annotation/annotation/gradle.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/annotation/annotation/gradle.properties b/annotation/annotation/gradle.properties
new file mode 100644
--- /dev/null	(revision 566a9f8d01b29a076829c9cf2d59e1fb54408980)
+++ b/annotation/annotation/gradle.properties	(revision 566a9f8d01b29a076829c9cf2d59e1fb54408980)
@@ -0,0 +1,1 @@
+androidx.subVersionCheckEnabled=true
\ No newline at end of file
Index: libraryversions.toml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libraryversions.toml b/libraryversions.toml
--- a/libraryversions.toml	(revision 2f9bf2b8adda49cb44020bd7ea671eb78cd3ca7b)
+++ b/libraryversions.toml	(revision 2eb64f15e98c0135fa177dbbd9efef18ee59c830)
@@ -1,7 +1,7 @@
 [versions]
 ACTIVITY = "1.10.0-alpha01"
-ANNOTATION = "1.8.0"
-ANNOTATION_EXPERIMENTAL = "1.4.0-rc01"
+ANNOTATION = "1.8.0-KBA-001"
+ANNOTATION_EXPERIMENTAL = "1.4.0-KBA-001"
 APPACTIONS_BUILTINTYPES = "1.0.0-alpha01"
 APPACTIONS_INTERACTION = "1.0.0-alpha01"
 APPCOMPAT = "1.7.0-alpha04"
Index: gradle/verification-metadata.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gradle/verification-metadata.xml b/gradle/verification-metadata.xml
--- a/gradle/verification-metadata.xml	(revision 566a9f8d01b29a076829c9cf2d59e1fb54408980)
+++ b/gradle/verification-metadata.xml	(revision 2eb64f15e98c0135fa177dbbd9efef18ee59c830)
@@ -25,6 +25,9 @@
          <trust file=".*[.]asc" regex="true"/>
          <trust group="^androidx(?!\.compose.compiler\b)\..*" regex="true" reason="not signed yet"/>
          <trust group="^com[.]android($|([.].*))" regex="true" reason="b/215430394"/>
+         <trust group="org.jetbrains.kotlin"/>
+         <trust group="org.jetbrains.kotlinx"/>
+         <trust group="org.jetbrains.kotlin.jvm"/>
       </trusted-artifacts>
       <trusted-keys>
          <trusted-key id="00089EE8C3AFA95A854D0F1DF800DD0933ECF7F7" group="com.google.guava"/>
Index: annotation/annotation-experimental/gradle.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/annotation/annotation-experimental/gradle.properties b/annotation/annotation-experimental/gradle.properties
new file mode 100644
--- /dev/null	(revision d0aefb8c2a415d334c12b2dd41dbd9d931ec6b3d)
+++ b/annotation/annotation-experimental/gradle.properties	(revision d0aefb8c2a415d334c12b2dd41dbd9d931ec6b3d)
@@ -0,0 +1,1 @@
+androidx.subVersionCheckEnabled=true
\ No newline at end of file
Index: buildSrc/private/src/main/kotlin/androidx/build/AndroidXPlaygroundRootImplPlugin.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXPlaygroundRootImplPlugin.kt b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXPlaygroundRootImplPlugin.kt
--- a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXPlaygroundRootImplPlugin.kt	(revision 31e945d64cd4ca1e0977c8c5b7d5468b891b860a)
+++ b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXPlaygroundRootImplPlugin.kt	(revision 3b23fa585e65158101d741ea69b9428284608296)
@@ -142,6 +142,9 @@
         google()
         mavenCentral()
         gradlePluginPortal()
+        maven {
+            it.url = URI("https://mirrors.tencent.com/nexus/repository/maven-public")
+        }
     }
 
     private class PlaygroundRepositories(props: PlaygroundProperties) {
Index: buildSrc/private/src/main/kotlin/androidx/build/clang/KonanBuildService.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/private/src/main/kotlin/androidx/build/clang/KonanBuildService.kt b/buildSrc/private/src/main/kotlin/androidx/build/clang/KonanBuildService.kt
--- a/buildSrc/private/src/main/kotlin/androidx/build/clang/KonanBuildService.kt	(revision 31e945d64cd4ca1e0977c8c5b7d5468b891b860a)
+++ b/buildSrc/private/src/main/kotlin/androidx/build/clang/KonanBuildService.kt	(revision 2eb64f15e98c0135fa177dbbd9efef18ee59c830)
@@ -132,51 +132,51 @@
      * @see ClangSharedLibraryTask
      */
     fun createSharedLibrary(parameters: ClangSharedLibraryParameters) {
-        val outputFile = parameters.outputFile.get().asFile
-        outputFile.delete()
-        outputFile.parentFile.mkdirs()
-
-        val platform = getPlatform(parameters.konanTarget)
-
-        // Specify max-page-size to align ELF regions to 16kb
-        val linkerFlags = if (parameters.konanTarget.get().asKonanTarget.family == Family.ANDROID) {
-            listOf("-z", "max-page-size=16384")
-        } else emptyList()
-
-        val objectFiles = parameters.objectFiles.regularFilePaths()
-        val linkedObjectFiles = parameters.linkedObjects.regularFilePaths()
-        val linkCommands = platform.linker.finalLinkCommands(
-            objectFiles = objectFiles,
-            executable = outputFile.canonicalPath,
-            libraries = linkedObjectFiles,
-            linkerArgs = linkerFlags,
-            optimize = true,
-            debug = false,
-            kind = LinkerOutputKind.DYNAMIC_LIBRARY,
-            outputDsymBundle = "unused",
-            needsProfileLibrary = false,
-            mimallocEnabled = false,
-            sanitizer = null
-        )
-        linkCommands.map { it.argsWithExecutable }.forEach { args ->
-            execOperations.executeSilently { execSpec ->
-                execSpec.executable = args.first()
-                args.drop(1).filterNot {
-                    // TODO b/305804211 Figure out if we would rather pass all args manually
-                    // We use the linker that konan uses to be as similar as possible but that
-                    // linker also has konan demangling, which we don't need and not even available
-                    // in the default distribution. Hence we remove that parameters.
-                    // In the future, we can consider not using the `platform.linker` but then
-                    // we would need to parse the konan.properties file to get the relevant
-                    // necessary parameters like sysroot etc.
-                    // https://github.com/JetBrains/kotlin/blob/master/kotlin-native/build-tools/src/main/kotlin/org/jetbrains/kotlin/KotlinNativeTest.kt#L536
-                    it.contains("--defsym") ||
-                        it.contains("Konan_cxa_demangle")
-                }.forEach {
-                    execSpec.args(it)
-                }
-            }
-        }
+//        val outputFile = parameters.outputFile.get().asFile
+//        outputFile.delete()
+//        outputFile.parentFile.mkdirs()
+//
+//        val platform = getPlatform(parameters.konanTarget)
+//
+//        // Specify max-page-size to align ELF regions to 16kb
+//        val linkerFlags = if (parameters.konanTarget.get().asKonanTarget.family == Family.ANDROID) {
+//            listOf("-z", "max-page-size=16384")
+//        } else emptyList()
+//
+//        val objectFiles = parameters.objectFiles.regularFilePaths()
+//        val linkedObjectFiles = parameters.linkedObjects.regularFilePaths()
+//        val linkCommands = platform.linker.finalLinkCommands(
+//            objectFiles = objectFiles,
+//            executable = outputFile.canonicalPath,
+//            libraries = linkedObjectFiles,
+//            linkerArgs = linkerFlags,
+//            optimize = true,
+//            debug = false,
+//            kind = LinkerOutputKind.DYNAMIC_LIBRARY,
+//            outputDsymBundle = "unused",
+//            needsProfileLibrary = false,
+//            mimallocEnabled = false,
+//            sanitizer = null
+//        )
+//        linkCommands.map { it.argsWithExecutable }.forEach { args ->
+//            execOperations.executeSilently { execSpec ->
+//                execSpec.executable = args.first()
+//                args.drop(1).filterNot {
+//                    // TODO b/305804211 Figure out if we would rather pass all args manually
+//                    // We use the linker that konan uses to be as similar as possible but that
+//                    // linker also has konan demangling, which we don't need and not even available
+//                    // in the default distribution. Hence we remove that parameters.
+//                    // In the future, we can consider not using the `platform.linker` but then
+//                    // we would need to parse the konan.properties file to get the relevant
+//                    // necessary parameters like sysroot etc.
+//                    // https://github.com/JetBrains/kotlin/blob/master/kotlin-native/build-tools/src/main/kotlin/org/jetbrains/kotlin/KotlinNativeTest.kt#L536
+//                    it.contains("--defsym") ||
+//                        it.contains("Konan_cxa_demangle")
+//                }.forEach {
+//                    execSpec.args(it)
+//                }
+//            }
+//        }
     }
 
     private fun FileCollection.regularFilePaths(): List<String> {
Index: buildSrc/shared.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/shared.gradle b/buildSrc/shared.gradle
--- a/buildSrc/shared.gradle	(revision 31e945d64cd4ca1e0977c8c5b7d5468b891b860a)
+++ b/buildSrc/shared.gradle	(revision 2eb64f15e98c0135fa177dbbd9efef18ee59c830)
@@ -34,7 +34,7 @@
     kotlinOptions {
         jvmTarget = "17"
         freeCompilerArgs += [
-                "-Werror",
+//                "-Werror",
                 "-Xskip-metadata-version-check",
                 "-Xjdk-release=17",
         ]
Index: playground-common/configure-plugin-management.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/playground-common/configure-plugin-management.gradle b/playground-common/configure-plugin-management.gradle
--- a/playground-common/configure-plugin-management.gradle	(revision 31e945d64cd4ca1e0977c8c5b7d5468b891b860a)
+++ b/playground-common/configure-plugin-management.gradle	(revision 3b23fa585e65158101d741ea69b9428284608296)
@@ -8,4 +8,7 @@
     mavenCentral()
     google()
     gradlePluginPortal()
+    maven {
+        url = "https://mirrors.tencent.com/nexus/repository/maven-public"
+    }
 }
